src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -197,6 +204,49 @@ protected static function bootTraits()
         }
     }
 
+    /**
+     * Disables relationship model touching for the current class during given callback scope.
+     *
+     * @param $callback
+     */
+    public static function withoutTouching($callback)
+    {
+        static::withoutTouchingOn([static::class], $callback);
+    }
+
+    /**
+     * Disables relationship model touching for the given model classes during given callback scope.
+     *
+     * @param array $models
+     * @param $callback
+     */
+    public static function withoutTouchingOn(array $models, $callback)
+    {
+        static::$ignoreOnTouch = array_values(array_merge(static::$ignoreOnTouch, $models));
+
+        try {
+            call_user_func($callback);
+        } finally {
+            static::$ignoreOnTouch = array_values(array_diff(static::$ignoreOnTouch, $models));
+        }
+    }
+
+    /**
+     * Whether or not the current model is going to be touched.
+     *
+     * @return bool
+     */
+    public function shouldTouch()
+    {
+        foreach (static::$ignoreOnTouch as $ignoredClass) {
+            if ($this instanceof $ignoredClass) {
```

Comment:
I think this could cause some surprise to users that extend models. I think it would be better to check `if(get_class($this) === $ignoredClass)`

Reply:
I would disagree, I would be surprised if extending an ignored class causes the child class to not be ignored.
Also, this allows you to have a parent "IgnoredModel" class and extend all your models from that instead of having to explicitly ignore every model.

Reply:
Yeah, this also allows us to globally ignore relationship touching by invoking it from the base model class. That's the desired behavior.